lib/bolt/transport/ssh.rb

```diff
@@ -138,7 +140,12 @@ def run_task(target, task, arguments, options = {})
             end
 
             conn.with_remote_tempdir do |dir|
-              remote_task_path = conn.write_remote_executable(dir, executable)
+              if task.respond_to? :file_content
```

Comment:
This probably breaks `apply`. Apply would pass in an object that responds to `file_content`, but `file_content` returns nil. So you should also check that `file_content` is not nil.

Reply:
Eventually it'd probably make sense to wrap the Puppet anon classes we get for tasks in our own task object so we can have a consistent API.